elixir/lib/reducer.ex

```diff
@@ -3,9 +3,12 @@ defmodule Reducer do
 
   def reduce(mappings, destination) do
     reduction = Enum.reduce(mappings, HashDict.new, &reduce_mappings/2)
-    final = HashDict.to_list(reduction)
```

Comment:
A HashDict is an enumerable so we don't need to convert it before calling `Enum.sort/1`.